### PR TITLE
[INTER-19] SPA improvements

### PR DESCRIPTION
**Pull request by TheUnderScorer**
This PR introduces suggested improvements from the related issue, except for this one:
> Maybe a getter/method like cacheDepth for getting the number of cached fp identifications that are on the client..

because it would mean we had to introduce breaking changes to the cache interface. If we still want to introduce it, we'll need to discuss the implementation first.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -50,7 +50,9 @@ export interface CustomAgent {
   load: (options: FingerprintJS.LoadOptions) => Promise<FingerprintJS.Agent>
 }
 
-export interface FpjsSpaOptions extends FpjsClientOptions {
+export interface FpjsSpaOptions
+  extends Omit<FpjsClientOptions, 'loadOptions'>,
```
Comment (ilfa) [inline src/client.ts:54]: It was a complicated way to make `loadOptions` optional!

File: README.md [hunk 2]
```diff
@@ -113,14 +113,24 @@ Before you start making identification requests to the Fingerprint Pro API, you
...
+
 const visitorData = await fpjsClient.getVisitorData()
 
 // with promises
 const visitorData = fpjsClient.init().then(() => {
   return fpjsClient.getVisitorData()
 })
 ```
-
+:You can also pass the load options here
```
Comment (JuroUhlar) [inline README.md:124]:
```suggestion
You can also pass the `loadOptions` into the `init` method here. They will be merged with the options passed to the constructor.  
```

**Comment by JuroUhlar** [comment]
> Made as we discussed, looks good. @JuroUhlar what do you think about readme?

It looks good! I made one suggestion. 
I also noticed this PR too late and merged my other readme PR, creating merge conflicts, sorry about that. @TheUnderScorer I can resolve the conflicts in this PR in you like, can I do that?

**Comment by TheUnderScorer** [comment]
Sure @JuroUhlar, feel free to resolve the conflicts, thanks! 😄

File: src/client.ts (hunk 1 above)
Comment (TheUnderScorer) [reply src/client.ts]: True 😅. I've simplified it a bit!

**Comment by JuroUhlar** [comment]
> Sure @JuroUhlar, feel free to resolve the conflicts, thanks! 😄

@TheUnderScorer Done!

**Review by JuroUhlar — APPROVED** [review]

**Review by ilfa — APPROVED** [review]

**Pull request merged**